sky/packages/sky/lib/src/fn3/homogeneous_viewport.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:sky/rendering.dart';
+import 'package:sky/src/fn3/framework.dart';
+import 'package:sky/src/fn3/basic.dart';
+
+typedef List<Widget> ListBuilder(int startIndex, int count);
```

Comment:
You want to pass the BuildContext here otherwise the callee can't access the theme and such.